homeassistant/components/letpot/select.py

```diff
@@ -0,0 +1,166 @@
+"""Support for LetPot select entities."""
+
+from collections.abc import Callable, Coroutine
+from dataclasses import dataclass
+from enum import StrEnum
+from typing import Any
+
+from letpot.deviceclient import LetPotDeviceClient
+from letpot.models import DeviceFeature, LightMode, TemperatureUnit
+
+from homeassistant.components.select import SelectEntity, SelectEntityDescription
+from homeassistant.const import EntityCategory
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddConfigEntryEntitiesCallback
+
+from .coordinator import LetPotConfigEntry, LetPotDeviceCoordinator
+from .entity import LetPotEntity, LetPotEntityDescription, exception_handler
+
+# Each change pushes a 'full' device status with the change. The library will cache
+# pending changes to avoid overwriting, but try to avoid a lot of parallelism.
+PARALLEL_UPDATES = 1
+
+
+class LightBrightnessLowHigh(StrEnum):
+    """Light brightness low/high model."""
+
+    LOW = "low"
+    HIGH = "high"
+
+
+def _get_brightness_low_high_value(coordinator: LetPotDeviceCoordinator) -> str | None:
+    """Return brightness as low/high for a device which only has a low and high value."""
+    brightness = coordinator.data.light_brightness
+    levels = coordinator.device_client.get_light_brightness_levels(
+        coordinator.device.serial_number
+    )
+    return (
+        LightBrightnessLowHigh.LOW.value
+        if levels[0] == brightness
+        else LightBrightnessLowHigh.HIGH.value
+    )
+
+
+async def _set_brightness_low_high_value(
+    device_client: LetPotDeviceClient, serial: str, option: str
+) -> None:
+    """Set brightness from low/high for a device which only has a low and high value."""
+    levels = device_client.get_light_brightness_levels(serial)
+    await device_client.set_light_brightness(
+        serial, levels[0] if option == LightBrightnessLowHigh.LOW.value else levels[1]
+    )
+
+
+@dataclass(frozen=True, kw_only=True)
+class LetPotSelectEntityDescription(LetPotEntityDescription, SelectEntityDescription):
+    """Describes a LetPot select entity."""
+
+    options_fn: Callable[[LetPotDeviceCoordinator], list[str]]
```

Comment:
I am not sure why `options_fn` is a function as you never really use the function but you just provide a static list in a lambda

Reply:
I used to have a function that did more, but decided to change that to a number entity and never reverted back to the normal `options`. Changed.